aws/s3/s3.go

```diff
@@ -0,0 +1,369 @@
+// Package s3 is for working with AWS S3 buckets.
+package s3
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/aws/retry"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/feature/s3/manager"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go-v2/service/s3/types"
+	"github.com/pkg/errors"
+)
+
+type S3 struct {
+	client     *s3.Client
+	uploader   *manager.Uploader
+	downloader *manager.Downloader
+}
+
+type Meta map[string]string
+
+// New returns an S3 struct which wraps an S3 client using the default AWS credentials chain.
+// This consults (in order) environment vars, config files, EC2 and ECS roles.
+// It is an error if the AWS_REGION environment variable is not set.
+// Requests with recoverable errors will be retried with the default retrier.
+func New() (S3, error) {
+	cfg, err := getConfig()
+	if err != nil {
+		return S3{}, errors.WithStack(err)
```

Comment:
Is the WithStack to form a boundary between the the errors related to the wrapper vs, those related to the underlying aws packages?

Reply:
Decided we don't need this, can just return `err`